### PR TITLE
ThreatQ Enhancements

### PR DESCRIPTION
**Pull request by orenzohar**
## Status
In Progress

## Related Issues
fixes: https://github.com/demisto/etc/issues/20539

## Description
add TLP values to outputs. also add advanced search and now supports multiple result from API

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [x] Documentation (with link to it)
- [ ] Code Review

## Technical writer review
Mention and link to the files that require a technical writer review.
- [ ] [YAML file](link)
- [ ] [CHANGELOG](link)

### CODE REVIEW

**Comment by kirbles19** [comment]
@orenzohar Done.

File: Integrations/ThreatQ_v2/ThreatQ_v2.py [hunk 1]
```diff
@@ -552,16 +561,28 @@ def add_malicious_data(generic_context, tq_score):
...
+        if dbot.get('Score') == 3:
+            add_malicious_data(generic, raw.get('TQScore', -1))
+
+        ec[outputPaths[indicator_type]].append(generic)
+        ec['DBotScore'].append(dbot)
+        if raw:
+            ec[CONTEXT_PATH['indicator']].append(raw)
+
+    # backwards compatibility
+    ec['DBotScore'] = ec['DBotScore'][0] if len(ec['DBotScore']) == 1 else ec['DBotScore']
```
Comment (yuvalbenshalom) [inline Integrations/ThreatQ_v2/ThreatQ_v2.py]: is that necessary? please re check.

File: Integrations/ThreatQ_v2/ThreatQ_v2.py [hunk 2]
```diff
@@ -552,16 +561,28 @@ def add_malicious_data(generic_context, tq_score):
     }
 
 
-def set_indicator_entry_context(indicator_type, raw, dbot, generic):
-    if dbot.get('Score') == 3:
-        add_malicious_data(generic, raw.get('TQScore', -1))
+def set_indicator_entry_context(indicator_type, raws, dbots, generic):
```
Comment (yuvalbenshalom) [inline Integrations/ThreatQ_v2/ThreatQ_v2.py]: there is an implicit assumption that in every index in raw there is the corresponding value in dbots. I think this is confusing. better just pass an list of tuples.

File: Integrations/ThreatQ_v2/ThreatQ_v2.py [hunk 3]
```diff
@@ -617,6 +662,81 @@ def test_module():
...
+    )
+    try:
+        indicator_types = indicator_types_res.json().get('data')
+        for indicator in indicator_types:
+            if indicator.get('name', '').lower() == indicator_name.lower():
+                return indicator.get('id')
+
+        raise ValueError(f'Could not find indicator')
+    except ValueError:
+        raise ValueError(f'Could not parse data from ThreatQ [Status code: {indicator_types_res.status_code}]')
```
Comment (yuvalbenshalom) [inline Integrations/ThreatQ_v2/ThreatQ_v2.py]: add content text if exists to the error.

File: Integrations/ThreatQ_v2/ThreatQ_v2.py [hunk 4]
```diff
@@ -617,6 +662,81 @@ def test_module():
...
+    res = tq_request(
+        method='POST',
+        url_suffix=f'/search/advanced?limit={limit}',
+        params=search_body,
+        retrieve_entire_response=True
+    )
+    try:
+        search_results = res.json().get('data')
+    except ValueError:
+        raise ValueError(f'Could not parse data from ThreatQ [Status code: {res.status_code}]')
```
Comment (yuvalbenshalom) [inline Integrations/ThreatQ_v2/ThreatQ_v2.py]: add content text if exists to the error.

**Comment by orenzohar** [comment]
@yuvalbenshalom I still need to run the test before I merge. waiting for dan to finish his updated test

**Comment by orenzohar** [comment]
passed both the old test and the new one 
image

**Comment by orenzohar** [comment]
@yuvalbenshalom two unrelated errors:
- `Impossible Traveler - Test Failed to run.`
- `Failed: playbook: Test Convert file hash to corresponding hashes with integration(s): VirusTotal failed.`

**Comment by kirbles19** [comment]
@orenzohar Can you specify what the TLP values are and exactly where they were added? Seems like something we can add manually, won't require new docs.

**Comment by dantavori** [comment]
@kirbles19 Traffic Light Protocol (TLP) schema provides a set of designations used to ensure that sensitive information is shared with the appropriate audience. ThreatQ provides a method for designating the availability of intelligence information by their sources.

The optional values are **Red** (not for disclosure, restricted to participants only), **Amber** (limited disclosure, restricted to participant's organizations), **Green** (limited disclosure, restricted to the community), **White** (disclosure is not limited) or **None** (default).

In ThreatQ UI, sources' TLP values can be modified from Settings > Data Management > TLP. I don't think Oren implemented a way to set a TLP value of a source from Demisto, but if it's already set then it's supposed to be under `ThreatQ.<object_type>.Source.TLP`.

**Pull request merged**